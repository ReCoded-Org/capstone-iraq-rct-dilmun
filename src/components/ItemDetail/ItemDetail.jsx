import React from 'react'
import PropTypes from 'prop-types'

import { useTranslation } from 'react-i18next'

export default function ProductDetail({ title, price, description }) {
    const { t } = useTranslation()
    return (
        <>
            <b className="block m-1">{title}</b>
            <p className="bg-yellow p-1 m-1 text-black inline-block">
                {t('proudctDetail.price')}: $ {price}
            </p>
            <b className="block m-1 mt-3"> {t('proudctDetail.description')}</b>
            <p className=" m-1 text-black ">{description}</p>
        </>
    )
}

ProductDetail.propTypes = {
    title: PropTypes.string,
    price: PropTypes.number,
    description: PropTypes.string,
}

ProductDetail.defaultProps = {
    title: '',
    price: '',
    description: '',
}
