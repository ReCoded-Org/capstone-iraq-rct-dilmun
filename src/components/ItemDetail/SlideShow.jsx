import React from 'react'
import { Carousel } from 'react-responsive-carousel'
import 'react-responsive-carousel/lib/styles/carousel.min.css'

import Img1 from '../../assets/classy.jpg'
import Img2 from '../../assets/profile_bg.png'

export default function SlideShow() {
    const myImgs = [Img1, Img2]

    return (
        <div>
            <div className="grid gric grid-cols-1 lg:grid-cols-12  ">
                <Carousel
                    showArrows
                    showThumbs
                    className="lg:col-span-8 lg:col-start-2  "
                >
                    {myImgs.map((img) => {
                        return (
                            <div className="border-2 border-grey rounded-xl  overflow-hidden p-1  ">
                                <img
                                    alt=""
                                    src={img}
                                    className="  object-contain max-h-96 w-full "
                                />
                            </div>
                        )
                    })}
                </Carousel>
            </div>
        </div>
    )
}
