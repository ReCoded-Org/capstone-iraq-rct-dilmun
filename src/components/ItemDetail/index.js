export { default } from './ItemDetail'
export { default as SlideShow } from './SlideShow'
