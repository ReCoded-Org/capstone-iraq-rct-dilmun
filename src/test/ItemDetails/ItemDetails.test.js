import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import ItemDetail from '../../components/ItemDetail'

it('Test: Item details without any props', () => {
    const tree = renderer
        .create(
            <Router>
                <ItemDetail />
            </Router>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})

it('Test: Item details with props', () => {
    const tree = renderer
        .create(
            <Router>
                <ItemDetail price={10} title="hey" description="hello" />
            </Router>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})
