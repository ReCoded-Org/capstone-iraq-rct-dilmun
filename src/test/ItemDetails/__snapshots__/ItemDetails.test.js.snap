// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test: Item details with props 1`] = `
Array [
  <b
    className="block m-1"
  >
    hey
  </b>,
  <p
    className="bg-yellow p-1 m-1 text-black inline-block"
  >
    proudctDetail.price
    : $ 
    10
  </p>,
  <b
    className="block m-1 mt-3"
  >
     
    proudctDetail.description
  </b>,
  <p
    className=" m-1 text-black "
  >
    hello
  </p>,
]
`;

exports[`Test: Item details without any props 1`] = `
Array [
  <b
    className="block m-1"
  >
    
  </b>,
  <p
    className="bg-yellow p-1 m-1 text-black inline-block"
  >
    proudctDetail.price
    : $ 
    0
  </p>,
  <b
    className="block m-1 mt-3"
  >
     
    proudctDetail.description
  </b>,
  <p
    className=" m-1 text-black "
  >
    
  </p>,
]
`;
